src/codegen/properties/functions.py

```diff
@@ -0,0 +1,75 @@
+"""Structure for storing function metadata from scrapigen."""
+from codegen.properties.adaptor_parameter import AdaptorParameter
+from codegen.properties.function_parameters import FunctionsParameter
+from codegen.properties.parameter import Parameter
+
+
+class Functions:
+    """Structure for storing function metadata from scrapigen."""
+
+    def __init__(self, function_name, function_metadata):
+        """Structure for storing function metadata from scrapigen."""
+        self._function_name = function_name
+        self._c_function_name = function_metadata["c_function_name"]
+        self._description = function_metadata["'description'"]
+        self._is_python_factory = function_metadata.get("is_python_factory", False)
+        self._python_class_name = function_metadata["python_class_name"]
+        self._calling_convention = function_metadata["calling_convention"]
+        self._return_type = function_metadata["returns"]
+        if "handle_parameter" in function_metadata:
+            self._handle_parameters = []
+            for name, parameter_data in function_metadata["handle_parameters"].items():
+                self._handle_parameters.append(Parameter(name, parameter_data))
```

Comment:
`handle_parameter` is singular. It describes a single parameter. You can't build a list out of it.

```
    'CalculateReversePolyCoeff': {
        'calling_convention': 'StdCall',
        'handle_parameter': {
            'ctypes_data_type': 'ctypes.c_char_p',
            'cvi_name': 'name',
            'python_accessor': 'self._name'
        },
        'parameters': [
            {
                'direction': 'in',
```